apps/frontend/src/libs/components/quiz-categories-selection/quiz-categories-form.tsx

```diff
@@ -0,0 +1,83 @@
+import { useCallback } from "react";
+import { useForm } from "react-hook-form";
+
+import { Button } from "../components.js";
+import { useQuizCategories } from "./libs/hooks/hooks.js";
+import styles from "./styles.module.css";
+
+type FormFields = {
+	categoriesIds: string[];
+};
+
+type Properties = {
+	onSubmit?: (payload: FormFields) => void;
+	submitLabel: string;
+};
+
+const QuizCategoriesForm: React.FC<Properties> = ({
+	onSubmit = (): void => {},
+	submitLabel,
+}: Properties) => {
+	const { handleSubmit, register, setValue, watch } = useForm<FormFields>({
+		defaultValues: { categoriesIds: [] },
+	});
+
+	const { isLoading, quizCategories } = useQuizCategories();
+
+	const isAllInputsChecked =
+		watch("categoriesIds").length === quizCategories.length;
+
+	const onAllInputClick: () => void = useCallback(() => {
```

Comment:
A10

Reply:
Can you please clarify what `A10` refers to?

Reply:
This is a quality criteria. Here is the [link](https://github.com/BinaryStudioAcademy/quality-criteria/blob/production/src/javascript.md) or you can find it in readme.md for bebalance project

Reply:
Oh my thx, didn't notice it before